src/database/bplus/tree/api.rs

```diff
@@ -0,0 +1,60 @@
+//! # Bplus Tree API [WIP]
```

Comment:
sg!
what i was going for was just an bplus tree api for the database to use but
i am going to follow your recommendation because i am still familiarizing with modules, and it also does what i wanted the api.rs file to do anyways
i do wonder if i missed the point of separating into mod.rs and tree.rs, so please see if the new patchset contains what you have in mind